util/cron/nightly_email_file.pm

```diff
@@ -8,22 +8,26 @@ use nightlysubs;
 sub writeFile{
     $num_args = @_;
 
-    if ($num_args != 18) {
+    if ($num_args != 17) {
         print "usage: nightly_email.pm \$status \$rawsummary \$sortedsummary \n";
-        print "         \$prevsummary \$mysystemlog \$prevmysystemlog \$mailer \$nochangerecipient \$recipient \n";
+        print "         \$prevsummary \$mysystemlog \$prevmysystemlog \$nochangerecipient \$recipient \n";
         print "         \$subjectid \$config_name \$revision \$rawlog \$starttime \n";
         print "         \$endtime \$crontab \$testdirs \$debug\n";
+
+        print "INFO: Number of arguments is $num_args\n";
+        for my $i (0 .. $#_) {
+            print "INFO: Argument $i: $_[$i]\n";
+        }
         exit 1;
     }
-    my ($status, $rawsummary, $sortedsummary, ,$prevsummary, $mysystemlog, $prevmysystemlog, $mailer, $nochangerecipient, $recipient, $subjectid, $config_name, $revision, $rawlog, $starttime, $endtime, $crontab, $testdirs, $debug)=@_;
+    my ($status, $rawsummary, $sortedsummary, ,$prevsummary, $mysystemlog, $prevmysystemlog, $nochangerecipient, $recipient, $subjectid, $config_name, $revision, $rawlog, $starttime, $endtime, $crontab, $testdirs, $debug)=@_;
 
     $status = $_[0];
     $rawsummary = $_[1];
     $sortedsummary = $_[2];
     $prevsummary = $_[3];
     $mysystemlog = $_[4];
     $prevmysystemlog = $_[5];
-    $mailer = $_[6];
```

Comment:
I don't know perl but I think we have to change all the numbers after this one since they're positional arguments

Reply:
Ugh - Yes, yes we do... PUSHUPS!!!